### PR TITLE
Add: deadlock check

### PR DESCRIPTION
**Pull request by AlexStocks**
Add a deadlock check lib

### CODE REVIEW

File: strings/slice_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	// 编译器会把 "hello, world" 这个字符串常量的字节数组分配在没有写权限的 memory segment
+	a := "hello world"
+	b := Slice(a)
+
+	// !!! 上面这个崩溃在defer里面是recover不回来的，真的就崩溃了，原因可能就跟c的非法内存访问一样，os不跟你玩了
+	// b[0] = 'a' //这里就等着崩溃吧
+
+	//但是可以这样，因为go又重新给b分配了内存
+	b = append(b, "hello world"...)
+	println(String(b)) // output: hello worldhello world
```
Comment (wongoo) [inline strings/slice_test.go]: assert check

File: strings/slice_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package gxstrings
+
+import (
+	"testing"
```
Comment (wongoo) [inline strings/slice_test.go]: I'd like to know the performance between `Slice(s string) (b []byte)` and `[]byte(s)`, can you provide a bench test for it?

File: strings/slice.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2016 ~ 2018 AlexStocks(https://github.com/AlexStocks).
+// All rights reserved.  Use of this source code is
+// governed by Apache License 2.0.
+
+// http://blog.csdn.net/siddontang/article/details/23541587
+// reflect.StringHeader和reflect.SliceHeader的结构体只相差末尾一个字段(cap)
```
[the remark this answers is not in the document]
Comment (AlexStocks) [reply strings/slice.go]: done

File: strings/slice_test.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+// go test  -v slice_test.go  slice.go
+
+// slice转string之后，如果slice的值有变化，string也会跟着改变
+func TestString(t *testing.T) {
+	b := []byte("hello world")
+	a := String(b)
+	b[0] = 'a'
+	println(a) //output  aello world
```
[the remark this answers is not in the document]
Comment (AlexStocks) [reply strings/slice_test.go]: done

File: strings/slice_test.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func TestSlice(t *testing.T) {
+	// 编译器会把 "hello, world" 这个字符串常量的字节数组分配在没有写权限的 memory segment
```
[the remark this answers is not in the document]
Comment (AlexStocks) [reply strings/slice_test.go]: done

File: strings/slice_test.go [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	a := "hello world"
+	b := Slice(a)
+
+	// !!! 上面这个崩溃在defer里面是recover不回来的，真的就崩溃了，原因可能就跟c的非法内存访问一样，os不跟你玩了
```
[the remark this answers is not in the document]
Comment (AlexStocks) [reply strings/slice_test.go]: delete this useless test.

File: strings/slice_test.go (hunk 1 above)
Comment (AlexStocks) [reply strings/slice_test.go]: done.

File: strings/slice_test.go (hunk 2 above)
Comment (AlexStocks) [reply strings/slice_test.go]: ok

File: bytes/slice_pool_test.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+	return rand.Intn(n) + 1
+}
+
+func ExampleGetBytes() {
+	// Obtain a buffer from the pool.
+	bufPtr := GetBytes(16)
+	defer PutBytes(bufPtr)
+	buf := *bufPtr
+	copy(buf, []byte("hello, world"))
+	println(string(buf))
```
[the remark this answers is not in the document]
Comment (AlexStocks) [reply bytes/slice_pool_test.go]: bufPtr type is *[]byte, so buf type is []byte. i do not know why i have to check it again

[the remark this answers is not in the document]
Comment (wongoo) [reply bytes/slice_pool_test.go]: I mean change to `assert.Equal(t, "hello, world", string(buf))`

[the remark this answers is not in the document]
Comment (AlexStocks) [reply bytes/slice_pool_test.go]: it is a example func "ExampleGetBytes".  however，I check it by panic.

**Comment by wongoo** [comment]
LGTM , ready to being merged

**Pull request merged**